#!/usr/bin/env python
PACKAGE = "navigation"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("gate_safety_margin",       double_t,    0, "all gates are shrinked by this value", 0,  0, 0.1)
gen.add("vehicle_safety_rear_margin",    double_t,    0, "the vehicle is extended by this value", 0,  0, 0.1)
gen.add("vehicle_safety_front_margin",    double_t,    0, "the vehicle is extended by this value", 0,  0, 0.1)
gen.add("vehicle_safety_width_margin",    double_t,    0, "the vehicle is extended by this value", 0,  0, 0.1)
gen.add("obstacle_safety_margin",   double_t,    0, "all obstacles are extended by this value", 0,  0, 0.1)
gen.add("road_closure_safety_margin",   double_t,    0, "all road closures are extended by this value", 0,  0, 0.1)
gen.add("left_lane_block_distance_in_front_of_obstacle", double_t,    0, "maximal distance to an obstacle at which the second lane can be used in front of obstacle", 0,  0.1, 2)
gen.add("left_lane_block_distance_behind_obstacle", double_t,    0, "maximal distance to an obstacle at which the second lane can be used behind obstacle", 0,  0.1, 2)

gen.add("block_left_lane", bool_t,    0, "false if left lane can be used freely", True)
gen.add("avoid_obstacles", bool_t,    0, "false if obstacles should be ignored", True)

gen.add("large_shrink_distance_for_warnings",   double_t,    0, "threshold for shrink warnings", 0,  0.1, 2)

exit(gen.generate(PACKAGE, "navigation", "CollisionDetection"))
